app/src/test/java/org/simple/clinic/util/ErrorResolverTest.kt

```diff
@@ -0,0 +1,62 @@
+package org.simple.clinic.util
+
+import com.google.common.truth.Truth.assertThat
+import com.nhaarman.mockito_kotlin.mock
+import com.squareup.moshi.JsonDataException
+import io.reactivex.exceptions.CompositeException
+import junitparams.JUnitParamsRunner
+import junitparams.Parameters
+import okhttp3.internal.http2.ConnectionShutdownException
+import okhttp3.internal.http2.ErrorCode
+import okhttp3.internal.http2.StreamResetException
+import org.junit.Test
+import org.junit.runner.RunWith
+import retrofit2.HttpException
+import java.net.SocketException
+import java.net.SocketTimeoutException
+import java.net.UnknownHostException
+
+@RunWith(JUnitParamsRunner::class)
+class ErrorResolverTest {
+
+  @Test
+  @Parameters(method = "network related errors")
+  fun `network errors should be identified correctly`(error: Throwable) {
+    val resolvedError = ErrorResolver.resolve(error)
+    assertThat(resolvedError.shouldBeReported).isFalse()
+    assertThat(resolvedError).isInstanceOf(ResolvedError.NetworkRelated::class.java)
+  }
+
+  @Suppress("unused")
+  fun `network related errors`() = arrayOf<Any>(
+      SocketException(),
+      SocketTimeoutException(),
+      UnknownHostException(),
+      StreamResetException(ErrorCode.ENHANCE_YOUR_CALM),
```

Comment:
TIL this is an actual error in HTTP/2.

Reply:
It's a name for an unofficial 420 http code.